File: Module II/13. Databases/EXAM - DB/01. Code first/Superheroes/Superheroes.Models/Power.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Superheroes.Models
{
    public class Power
    {
        private ICollection<Superhero> superheros;

        public Power()
        {
            this.superheros = new HashSet<Superhero>();
        }

        public int Id { get; set; }

        [Index(IsUnique = true)]
        [MaxLength(35), MinLength(3)]
        [Required]
        public string Name { get; set; }

        public virtual ICollection<Superhero> Superheros
        {
            get
            {
                return this.superheros;
            }
            set
            {
                this.superheros = value;
            }
        }
    }
}

File: Module III/16.Node.Js/workshop-superheroes/superheroes/server/models/power.js
"use strict";

const mongoose = require("mongoose");

const Schema = mongoose.Schema;

let powerSchema = new Schema({
    name: {
        type: String,
        required: true,
        minlength: 3, 
        maxlength: 35,
        unique: true
    },
    superheroes: [String]
});

const Power = mongoose.model("power", powerSchema);
module.exports = Power;